plugins/opercmds.py

```diff
@@ -26,29 +29,49 @@ def checkban(irc, source, args):
         irc.msg(source, "Checking for hosts that match \x02%s\x02:" % banmask, notice=True)
 
         results = 0
-        for uid, userobj in irc.users.copy().items():
-            if irc.match_host(banmask, uid):
-                if results < 50:  # XXX rather arbitrary limit
-                    s = "\x02%s\x02 (%s@%s) [%s] {\x02%s\x02}" % (userobj.nick, userobj.ident,
-                        userobj.host, userobj.realname, irc.get_friendly_name(irc.get_server(uid)))
-
-                    # Always reply in private to prevent information leaks.
-                    irc.reply(s, private=True)
-                results += 1
+        if options.channel:
+            if options.channel.startswith('#'):
```

Comment:
You should use a [custom argparse type](https://docs.python.org/3/library/argparse.html#type) here for channels, perhaps integrating it with the core of IRCParser entirely. Preferably, this filter should call `utils.isChannel` as well.